src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,24 @@
+package seedu.addressbook.data.person;
+
+public class Contact {
+    protected String value;
+    protected boolean isPrivate;
+    
+    public String toString() {
+        return value;
+    }
+
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof Address // instanceof handles nulls
```

Comment:
ouch! you see that you are comparing parent class with one other child class instance? think how you would alleviate this problem?